OCPBUGS#57348: release_notes/ocp-4-19-release-notes: Add a boot-image-clobber known issue

wking opened the pull request:
[Change management notification](https://mail.google.com/mail/u/0/?ik=a97decb9e4&view=om&permmsgid=msg-a:r-2688167191133413645) sent on the 3rd of July.

[OCPBUGS-57348](https://issues.redhat.com/browse/OCPBUGS-57348) is the 4.20 bug, and we don't have a 4.19.z clone yet, but we can update this entry once that 4.19.z OCPBUGS-... exists.

Version(s): 4.19

Issue: [OCPBUGS-57348](https://issues.redhat.com/browse/OCPBUGS-57348)

[Docs preview](https://94987--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-19-release-notes.html#ocp-4-19-known-issues_release-notes).

QE review:
- [x] QE has [approved](https://github.com/openshift/openshift-docs/pull/94987#issuecomment-3016300193) this change.

wking committed:
release_notes/ocp-4-19-release-notes: Add a boot-image-clobber known issue

wking commented:
@openshift/team-documentation, I'm not sure what the add-a-known-issue process is; can you help me get this ingested?

dfitzmau commented on release_notes/ocp-4-19-release-notes.adoc:
```diff
@@ -2741,6 +2741,8 @@ In the following tables, features are marked with the following statuses:
 
 * In {product-title} {product-version}, clusters using IPsec for network encryption might experience intermittent loss of pod-to-pod connectivity. This prevents some pods on certain nodes from reaching services on other nodes, resulting in connection timeouts. Internal testing could not reproduce this issue on clusters with 120 nodes or less. There is no workaround for this issue. (link:https://i ...
 
+* {product-title} clusters that are installed on {aws-short} with custom AMIs or on {gcp-short} with custom disk images will have those customizations overridden by boot image management. To recover, you must xref:../machine_configuration/mco-update-boot-images.adoc#mco-update-boot-images-disable_machine-configs-configure[disable boot image management], restore your MachineSet boot images, and de ...
```
```suggestion
* If you install a cluster on {aws-short} that has Amazon Machine Images (AMI) enabled or on  {gcp-short} that has custom disk images enabled, the boot image management overrides these customization images with boot images.  As a workaround, you can disable the boot image management feature, restore the boot images for the machine sets to their original location, and delete any machines that were incorrectly generated by the overriding boot images. (link:https://issues.redhat.com/browse/OCPBUGS-57348[OCPBUGS-57348])
```

wking replied:
I have some concerns with that suggestion, but in the interest of getting _something_ declared as a known-issue, I've adopted your wording (and pivoted from the 4.20 OCPBUGS-57348 to the 4.19.z OCPBUGS-57796, now that that backport tracker exists) with 72d6586371 -> cf256d9e38.  See the cf256d9e38 commit message for more on my concerns.  I'll leave it up to docs folks if any of them are worth follow-up pull requests.

/label merge-review-needed

wking committed:
release_notes/ocp-4-19-release-notes: Pick up Darragh's boot-image-clobber rewording

Pulling in the suggestion from [1], word for word (although I am
bumping the bug link to OCPBUGS-57796, now that that 4.19.z backport
tracker exists), to try to get something declared.  We can wordsmith
later if we want.  Personally, I have concerns, including:

> ... the boot image management overrides these customization images
> with boot images.

but "with boot images" doesn't make sense to me, because MachineSets
are going to reference boot images regardless; there's no "without
boot images" option.  The distinction is that sometimes the boot
images are specifically selected by the cluster admin, and the MCO's
boot image management would override those cluster-admin preferences.

> ...restore the boot images for the machine sets to their original
> location...

I'd prefer "previous value" or something to "original location".  I
haven't heard folks say "location" for an AMI ID or other MachineSet
property value, while I have heard "value" for that.  And it seems
like folks might confuse "original" as "what the MachineSet used when
it was created" when what we mean was "what the MachineSet used just
before the MCO clobbered its boot image value".

> ...delete any machines that were incorrectly generated by the
> overriding boot images.

I'd prefer "overriden boot images" or my "undesired boot images",
because the timeline there is:

1. MCO overrides the MachineSet's boot image configuration, inserting
   a stock boot image ID instead of the admin-preferred boot image ID.
2. Admin updates MachineConfiguration to disable MCO boot image
   management for that MachineSet.  After this, all overriding will be
   past-tense.  Cluster-admin preference for the admin-preferred boot
   image over the stock boot image can continue in the present-tense.
3. Admin restores MachineSet boot image ID configuration.
4. Admin deletes any Machines which launched from the stock boot
   image.

So I don't like the present-tense "overriding" in wording about step
(4).

It also feels weird to me to have:

> ...you can disable the boot image management feature...

as a non-link, with a later:

> To disable boot image management, see
> ref:../machine_configuration/mco-update-boot-images.adoc#mco-update-boot-images-disable_machine-configs-configure[disable
> boot image management].

giving the link.  I'd have expected some of the folks reading the
initial words to think "huh, how do I do this step?", and having the
words they were wondering about be the link to the docs that would
clear them up seems more usable than requiring them to skim to the end
of the paragraph to find the link they need.  But maybe there are doc
conventions around this whose motivation I don't understand?

Anyhow, none of my concerns seem large enough to be worth delaying a
known-issue declaration over, so I'm adopting Darragh's wording in
this commit, pointing out the places I don't agree with in this commit
message, and leaving it up to the docs folks to decide if any of my
concerns are worth further word-smithing in future pull requests.

[1]: https://github.com/openshift/openshift-docs/pull/94987#discussion_r2156495753

maxwelldb commented on release_notes/ocp-4-19-release-notes.adoc:
```diff
@@ -2741,6 +2741,8 @@ In the following tables, features are marked with the following statuses:
 
 * In {product-title} {product-version}, clusters using IPsec for network encryption might experience intermittent loss of pod-to-pod connectivity. This prevents some pods on certain nodes from reaching services on other nodes, resulting in connection timeouts. Internal testing could not reproduce this issue on clusters with 120 nodes or less. There is no workaround for this issue. (link:https://i ...
 
+* If you install a cluster on {aws-short} that has Amazon Machine Images (AMI) enabled or on  {gcp-short} that has custom disk images enabled, the boot image management overrides these customization images with boot images.  As a workaround, you can disable the boot image management feature, restore the boot images for the machine sets to their original location, and delete any machines that were ...
```
```suggestion
If you install a cluster on {aws-short} with Amazon Machine Images (AMI) enabled, or on {gcp-short} with custom disk images enabled, the boot image management feature overrides these custom images with default boot images.
+
As a workaround, you can disable the boot image management feature, restore the original boot images for the machine sets, and delete any machines that were incorrectly created by the overriding boot images.
+
To disable boot image management, see ref:../machine_configuration/mco-update-boot-images.adoc#mco-update-boot-images-disable_machine-configs-configure[Disable boot image management].
+
(link:https://issues.redhat.com/browse/OCPBUGS-57796[OCPBUGS-57796])
```

I'm thinking something like this? Be sure to remove doubled spaces and clarify the first clause, regardless.

wking replied:
> I'm thinking something like this?

Fine with me; pushed with cf256d9e38 -> a35fd43734.  I also have `Allow edits and access to secrets by maintainers` checked, so folks can push the rewords themselves, or close my pull and open your own?  I really don't care what the words are here, it just seems like an issue that is worth a known-issue declaration.  If there's a way I can delegate known-issue declaration to someone who has more time to shepherd the change through to landing, please let me know.

maxwelldb reviewed (commented):
One suggestion left. 

Because this impacts published release notes, it will need to go through the change management process. Be sure to indicate QE approval in the checkbox in the PR description, too. Thanks!

wking committed:
release_notes/ocp-4-19-release-notes: Pick up Max's boot-image-clobber rewording

Copy-pasted from Max's comment [1].

If we spend too long talking about the wording, the fix will be out
before we've warned anyone about the bug.

[1]: https://github.com/openshift/openshift-docs/pull/94987#discussion_r2166892245

gpei commented:
/lgtm

gpei commented:
@dfitzmau @maxwelldb Do you happen to know what is missing to merge this doc PR? The above information says that the status of the linked Jira issue is invalid, but it actually links to the Installer bug related to this PR, which was already verified and closed. Maybe we don't need a Jira issue to merge this at all, or there are other ways to manually merge the current PR. Thanks.

dfitzmau commented:
Hi @wking and @gpei . I send out a change management notifcation. We should give it a day or so to answer any responses and then we can get this PR merged. Thanks for your contribution!

gpei commented:
@dfitzmau thanks for the quick help on this!

dfitzmau commented:
> Hi @wking and @gpei . I send out a change management notifcation. We should give it a day or so to answer any responses and then we can get this PR merged. Thanks for your contribution!

Thanks, Jianli. I see that the 4.19.2 z-stream Images advisory lists the bug fix:

![Screenshot From 2025-07-03 12-10-26](https://github.com/user-attachments/assets/adf4ea86-f986-474c-9c6e-648ca5953641)

So the known issue will still exist for 4.19.0 and 4.19.1?
